### PR TITLE
docs: translate hooks-custom

### PR DESCRIPTION
**Pull request by neighborhood999**

### CODE REVIEW

File: content/docs/hooks-custom.md [hunk 1]
```diff
@@ -63,15 +63,15 @@ function FriendListItem(props) {
...
+相反的，我們想要在 `FriendStatus` 和 `FriendListItem` 共享這個邏輯。
 
-Traditionally in React, we've had two popular ways to share stateful logic between components: [render props](/docs/render-props.html) and [higher-order components](/docs/higher-order-components.html). We will now look at how Hooks solve many of the same problems without forcing you to add more components to the tree.
+習慣上，我們有兩種常用的方式在 component 來共享 stateful 邏輯：[render props](/docs/render-props.html) 和 [higher-order component](/docs/higher-order-components.html)。我們現在將看到 Hook 如何在不強迫 tree 加入更多 component 的情況下，解決許多相同的問題。
 
-## Extracting a Custom Hook {#extracting-a-custom-hook}
+## 提取一個自定義的 Hook {#extracting-a-custom-hook}
 
-When we want to share logic between two JavaScript functions, we extract it to a third function. Both components and Hooks are functions, so this works for them too!
+當我們想要共享邏輯在兩個 JavaScript function 之間時，我們提取它成為第三個 function。Componet 和 Hook 兩者都是 function，所以這也適用於它們！
```
Comment (chloewlin) [inline content/docs/hooks-custom.md]:
```suggestion
當我們想要共享邏輯在兩個 JavaScript function 之間時，我們提取它成為第三個 function。Component 和 Hook 兩者都是 function，所以這也適用於它們！
```

File: content/docs/hooks-custom.md [hunk 2]
```diff
@@ -184,24 +184,24 @@ function ChatRecipientPicker() {
...
   const isRecipientOnline = useFriendStatus(recipientID);
 ```
 
-This lets us know whether the *currently selected* friend is online. If we pick a different friend and update the `recipientID` state variable, our `useFriendStatus` Hook will unsubscribe from the previously selected friend, and subscribe to the status of the newly selected one.
+這讓我們知道*目前選擇的*朋友是否在線上。如果我們選擇不同的朋友並更新 `recipientID` state 變數，我們的 `useFriendStatus` Hook 將會從先前選擇的朋友中取消訂閱，並訂閱最新選擇的狀態。
 
 ## `useYourImagination()` {#useyourimagination}
 
-Custom Hooks offer the flexibility of sharing logic that wasn't possible in React components before. You can write custom Hooks that cover a wide range of use cases like form handling, animation, declarative subscriptions, timers, and probably many more we haven't considered. What's more, you can build Hooks that are just as easy to use as React's built-in features.
+自定義的 Hook 提供了共享邏輯的靈活性，這在以前的 React component 是不可能的。你可以撰寫自定義的 Hook 涵蓋廣泛的場景，想是表格處理、動畫、陳述式訂閱（Declarative Subscription）、計時器還有更多我們沒有考慮過的。更重要的是，你可以打造與 React 的內建一樣易於使用的 Hook。
```
Comment (chloewlin) [inline content/docs/hooks-custom.md]:
```suggestion
自定義的 Hook 提供了共享邏輯的靈活性，這在以前的 React component 是不可能的。你可以撰寫自定義的 Hook 涵蓋廣泛的場景，想是表格處理、動畫、陳述式訂閱（Declarative subscription）、計時器還有更多我們沒有考慮過的。更重要的是，你可以打造與 React 的內建一樣易於使用的 Hook。
```

File: content/docs/hooks-custom.md [hunk 3]
```diff
@@ -184,24 +184,24 @@ function ChatRecipientPicker() {
...
 
-Try to resist adding abstraction too early. Now that function components can do more, it's likely that the average function component in your codebase will become longer. This is normal -- don't feel like you *have to* immediately split it into Hooks. But we also encourage you to start spotting cases where a custom Hook could hide complex logic behind a simple interface, or help untangle a messy component.
+盡量不要過早地加入抽象。現在 function component 可以做更多的事，在你 codebase 中的 function componnet 程式碼平均可能都會變得更長。這都是正常的 -- 不要覺得你*必須*馬上把它拆分成 Hook。但我們也鼓勵你開始發現自定義的 Hook 可以隱藏簡單 interface 背後的複雜邏輯情況，或者幫忙解開一個混亂的 component。
```
Comment (chloewlin) [inline content/docs/hooks-custom.md]:
```suggestion
盡量不要過早地加入抽象。現在 function component 可以做更多的事，在你 codebase 中的 function component 程式碼平均可能都會變得更長。這都是正常的 -- 不要覺得你*必須*馬上把它拆分成 Hook。但我們也鼓勵你開始發現自定義的 Hook 可以隱藏簡單 interface 背後的複雜邏輯情況，或者幫忙解開一個混亂的 component。
```

File: content/docs/hooks-custom.md (hunk 2 above)
Comment (neighborhood999) [reply content/docs/hooks-custom.md]: `Declarative Subscription` is better.

Comment (chloewlin) [reply content/docs/hooks-custom.md]: Ok!

**Pull request merged**